src/components/trail.js

```diff
@@ -72,268 +279,268 @@ AFRAME.registerComponent('trail', {
-      vertices[36 * i + 13] = currentPoint.center.y;
-      vertices[36 * i + 14] = currentPoint.center.z;
-
-      // Color
-      colors[48 * i + 16] = color.red;
-      colors[48 * i + 17] = color.green;
-      colors[48 * i + 18] = color.blue;
-      colors[48 * i + 19] = alpha * 0.2;
-
-      vertices[36 * i + 15] = currentPoint.top.x;
-      vertices[36 * i + 16] = currentPoint.top.y;
-      vertices[36 * i + 17] = currentPoint.top.z;
-
-      // Color
-      colors[48 * i + 20] = color.red;
-      colors[48 * i + 21] = color.green;
-      colors[48 * i + 22] = color.blue;
-      colors[48 * i + 23] = alpha;
-
-      vertices[36 * i + 18] = previousPoint.bottom.x;
-      vertices[36 * i + 19] = previousPoint.bottom.y;
-      vertices[36 * i + 20] = previousPoint.bottom.z;
-
-      // Color
-      colors[48 * i + 24] = color.red;
-      colors[48 * i + 25] = color.green;
-      colors[48 * i + 26] = color.blue;
-      colors[48 * i + 27] = 0.0;
-
-      vertices[36 * i + 21] = currentPoint.center.x;
-      vertices[36 * i + 22] = currentPoint.center.y;
-      vertices[36 * i + 23] = currentPoint.center.z;
-
-      // Color
-      colors[48 * i + 28] = color.red;
-      colors[48 * i + 29] = color.green;
-      colors[48 * i + 30] = color.blue;
-      colors[48 * i + 31] = alpha * 0.2;
-
-      vertices[36 * i + 24] = previousPoint.center.x;
-      vertices[36 * i + 25] = previousPoint.center.y;
-      vertices[36 * i + 26] = previousPoint.center.z;
-
-      // Color
-      colors[48 * i + 32] = color.red;
-      colors[48 * i + 33] = color.green;
-      colors[48 * i + 34] = color.blue;
-      colors[48 * i + 35] = previousAlpha * 0.2;
-
-      vertices[36 * i + 27] = previousPoint.bottom.x;
-      vertices[36 * i + 28] = previousPoint.bottom.y;
-      vertices[36 * i + 29] = previousPoint.bottom.z;
-
-      // Color
-      colors[48 * i + 36] = color.red;
-      colors[48 * i + 37] = color.green;
-      colors[48 * i + 38] = color.blue;
-      colors[48 * i + 39] = 0.0;
-
-      vertices[36 * i + 30] = currentPoint.bottom.x;
-      vertices[36 * i + 31] = currentPoint.bottom.y;
-      vertices[36 * i + 32] = currentPoint.bottom.z;
-
-      // Color
-      colors[48 * i + 40] = color.red;
-      colors[48 * i + 41] = color.green;
-      colors[48 * i + 42] = color.blue;
-      colors[48 * i + 43] = 0.0;
-
-      vertices[36 * i + 33] = currentPoint.center.x;
-      vertices[36 * i + 34] = currentPoint.center.y;
-      vertices[36 * i + 35] = currentPoint.center.z;
-
-      // Color
-      colors[48 * i + 44] = color.red;
-      colors[48 * i + 45] = color.green;
-      colors[48 * i + 46] = color.blue;
-      colors[48 * i + 47] = alpha * 0.2;
-
-      previousAlpha = alpha;
+  createNewNode: function () {
+    const saberObject = this.saberEl.object3D;
+
+    let newNode;
+
+    switch (this.trailType) {
+      case 'bright':
+        newNode = {
+          from: new THREE.Vector3(0, -0.5 + this.brightTrailWidth, 0),
```

Comment:
It's better to reuse THREE objects. Especially if the function would be running every frame.